PlaceRouteHierFlow/placer/Placer.cpp

```diff
@@ -714,30 +714,47 @@ void Placer::ReshapeSeqPairMap(std::map<double, std::pair<SeqPair, ILP_solver>>&
   if(it!=Map.end()) {Map.erase(it, Map.end());}
 }
 
-void Placer::PlacementRegularAspectRatio_ILP(std::vector<PnRDB::hierNode>& nodeVec, string opath, int effort, PnRDB::Drc_info& drcInfo){
-  int nodeSize=nodeVec.size();
-  //cout<<"Placer-Info: place "<<nodeVec.back().name<<" in aspect ratio mode "<<endl;
+void Placer::PlacementRegularAspectRatio_ILP_core(int seed, design designData, int effort, PnRDB::Drc_info &drcInfo, int nodeSize, std::map<double, std::pair<SeqPair, ILP_solver>> &spVec){
   #ifdef RFLAG
   //cout<<"Placer-Info: run in random mode..."<<endl;
   srand (time(NULL));
   #endif
   #ifndef RFLAG
   //cout<<"Placer-Info: run in normal mode..."<<endl;
-  srand(0);
+  srand(seed);
   #endif
   int mode=0;
-  // Read design netlist and constraints
-  design designData(nodeVec.back());
-  designData.PrintDesign();
   // Initialize simulate annealing with initial solution
   SeqPair curr_sp(designData, size_t(1. * log(hyper.T_MIN/hyper.T_INT)/log(hyper.ALPHA) *
         ((effort == 0) ? 1. : ((effort == 1) ? 4. : 8.)) ));
   curr_sp.PrintSeqPair();
   ILP_solver curr_sol(designData);
-  std::map<double, std::pair<SeqPair, ILP_solver>> spVec=PlacementCoreAspectRatio_ILP(designData, curr_sp, curr_sol, mode, nodeSize, effort, drcInfo);
-  //curr_sol.updateTerminalCenter(designData, curr_sp);
-  //curr_sol.PlotPlacement(designData, curr_sp, opath+nodeVec.back().name+"opt.plt");
-  if((int)spVec.size()<nodeSize) {
+  //std::map<double, std::pair<SeqPair, ILP_solver>> spVec;
+  PlacementCoreAspectRatio_ILP(designData, curr_sp, curr_sol, mode, nodeSize, effort, drcInfo, spVec);
+}
+
+void Placer::PlacementRegularAspectRatio_ILP(std::vector<PnRDB::hierNode>& nodeVec, string opath, int effort, PnRDB::Drc_info& drcInfo){
+  int nodeSize=nodeVec.size();
+  //cout<<"Placer-Info: place "<<nodeVec.back().name<<" in aspect ratio mode "<<endl;
+  // Read design netlist and constraints
+  design designData(nodeVec.back());
+  designData.PrintDesign();
+  int n_threads = 1;
```

Comment:
@stevenmburns We can now specify the number of threads. Each thread uses a seed.